File: app/src/main/java/ru/strcss/projects/moneycalc/moneycalcandroid/componets/settings/SettingsContract.java
package ru.strcss.projects.moneycalc.moneycalcandroid.componets.settings;

import ru.strcss.projects.moneycalc.moneycalcandroid.BasePresenter;
import ru.strcss.projects.moneycalc.moneycalcandroid.BaseView;
import ru.strcss.projects.moneycalc.moneycalcdto.entities.SettingsLegacy;

/**
 * This specifies the contract between the view and the {@link SettingsPresenter}.
 */
public interface SettingsContract {
    interface View extends BaseView<Presenter> {
        void showUpdateSuccess();
    }

    interface Presenter extends BasePresenter<View> {
        void updateSettings(SettingsLegacy settings);
    }
}

File: app/src/main/java/ru/strcss/projects/moneycalc/moneycalcandroid/componets/settings/SettingsContract.kt
package ru.strcss.projects.moneycalc.moneycalcandroid.componets.settings

import ru.strcss.projects.moneycalc.moneycalcandroid.BasePresenter
import ru.strcss.projects.moneycalc.moneycalcandroid.BaseView
import ru.strcss.projects.moneycalc.moneycalcdto.entities.SettingsLegacy

/**
 * This specifies the contract between the view and the [SettingsPresenter].
 */
interface SettingsContract {
    interface View : BaseView<Presenter> {
        fun showUpdateSuccess()
    }

    interface Presenter : BasePresenter<View> {
        fun updateSettings(settings: SettingsLegacy)
    }
}
